### PR TITLE
Form checkbox

### PR DESCRIPTION
**Pull request by gaborszita**

### CODE REVIEW

**Comment by gaborszita** [comment]
In report:
image

Data aggregation:
image

File: app/src/screens/form-creation-flow/components/questions/Checkboxes.js [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+                    setAddOptionModalVisible(true);
+                  }}
+                  noDividerLine
+                />
+              </TouchableOpacity>
+              <TouchableOpacity
+                onLongPress={() => {
+                  setOptions(options.filter((_, i) => i !== index));
+                }}>
+                <Text
```
Comment (Versus7) [inline app/src/screens/form-creation-flow/components/questions/Checkboxes.js:92]: Bug: text not visible in dark mode.

File: app/src/components/form/Checkbox.js [hunk 2]
```diff
@@ -1,70 +1,55 @@
...
+      <Question title={title} />
+      <View
+        style={{
+          flex: 1,
+          flexDirection: 'column',
+          justifyContent: 'center',
+        }}>
+        {options.map((item, index) => {
+          return (
             <View
```
Comment (Versus7) [inline app/src/components/form/Checkbox.js:25]: Clicking anywhere in the row should toggle the checkbox, not just the checkbox itself.

File: app/src/components/form/Checkbox.js [hunk 3]
```diff
@@ -1,70 +1,55 @@
...
-        return (
-          <TouchableOpacity
-            disabled={disabled}
-            key={index}
-            onPress={() => {
-              onValueChange(option);
-            }}
-            style={{flexDirection: 'row', alignItems: 'center', margin: 10}}>
+    <View>
+      <Question title={title} />
```
Comment (Versus7) [inline app/src/components/form/Checkbox.js:16]: Should follow standard of other components and pass in a "onReset" function, that resets the checkboxes when question is long-pressed.

File: app/src/components/modals/ScoutViewer.js [hunk 4]
```diff
@@ -558,38 +559,25 @@ function ScoutViewer({
...
+                  {field.type === 'checkboxes' && (
+                    <View>
+                      <Checkbox
+                        title={field.question}
+                        options={field.options}
+                        value={tempData[index]}
+                        disabled={!editingActive}
+                        colors={colors}
+                        editingActive={editingActive}
+                        onValueChange={value => {
```
Comment (Versus7) [inline app/src/components/modals/ScoutViewer.js:571]: When testing this feature, it did not correctly save the different checkboxes for me.

File: app/src/screens/form-creation-flow/components/question-summaries/CheckboxesSummary.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
+import React from 'react-native';
+import SummaryTemplate from './SummaryTemplate';
+import SummaryText from './SummaryText';
+
+const CheckboxesSummary = ({question, onDelete}) => {
```
Comment (Versus7) [inline app/src/screens/form-creation-flow/components/question-summaries/CheckboxesSummary.js:5]: Have not tested since working off of local db.

**Review by Versus7 — CHANGES_REQUESTED** [review]

File: app/src/components/modals/ScoutViewer.js (hunk 4 above)
Comment (gaborszita) [reply app/src/components/modals/ScoutViewer.js:571]: it works on my computer

File: app/src/components/form/Checkbox.js (hunk 3 above)
Comment (gaborszita) [reply app/src/components/form/Checkbox.js:16]: fixed

**Pull request merged**